odin.md

```diff
@@ -519,42 +515,33 @@ point := [3]f32{10, 20, 1}
 transformed := transform * point      // Matrix multiplication
 
 // Quaternions for 3D rotations
-identity_rot := quaternion128{0, 0, 0, 1}  // No rotation
-rotation_90_z := quaternion128{0, 0, 0.707, 0.707}  // 90° around Z
+identity_rot := quaternion(w = 1, x = 0, y = 0, z = 0)  // No rotation
+rotation_90_z := quaternion(w = 0.707, x = 0, y = 0, z = 0.707)  // 90° around Z
+```
 
-////////////////////////////////////////////////////
 ## 14. Context System and Defer
-////////////////////////////////////////////////////
 
-// Odin has an implicit context system for threading allocators,
-// loggers, and other utilities through your program
+```
+// Odin has an implicit context system that makes it easy to use
+// temporary allocations in loops without manual cleanup
 
-example_with_context :: proc() {
-    // Save current context
-    old_allocator := context.allocator
-    
-    // Use a different allocator temporarily
-    temp_allocator := context.temp_allocator
-    context.allocator = temp_allocator
-    
-    // All allocations in this scope use temp_allocator
-    temp_data := make([]int, 100)
-    // No need to delete temp_data - it's automatically cleaned up
+process_files :: proc(filenames: []string) {
+    // Use temp allocator for temporary data in this scope
+    context.allocator = context.temp_allocator
+    defer free_all(context.temp_allocator)  // Clear the arena when done 
     
-    // Restore original allocator
-    context.allocator = old_allocator
+    for filename in filenames {
+        // Each iteration allocates temporary data
+        data := make([]u8, 1024)  // No defer is needed here
+        fmt.printf("Processing %s with %d bytes\n", filename, len(data))
+        // No individual cleanup needed
+    }
 }
```

Comment:
This is still a bad example of using the `context.temp_allocator`. Just use it directly and don't tell people to assign it to `context.allocator` as that will confuse them and cause them issues in the future if they think they must do that.

```odin
process_files :: proc(filenames: []string) {
    // Use temp allocator for temporary data in this scope
    defer free_all(context.temp_allocator)  // Clear the arena when done 
    
    for filename in filenames {
        // Each iteration allocates temporary data
        data := make([]u8, 1024, context.temp_allocator)  // No defer is needed here
        fmt.printf("Processing %s with %d bytes\n", filename, len(data))
        // No individual cleanup needed
    }
}
```